From 9fbaa8f5f1255e7383d25c03f40411c3d0fe6bc1 Mon Sep 17 00:00:00 2001
Message-Id: <9fbaa8f5f1255e7383d25c03f40411c3d0fe6bc1.1439343287.git.chang-joon.lee@intel.com>
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Tue, 28 Jul 2015 10:44:46 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: when scaling enabled check for
 scaled plane size

While scaling, instead of using gmch, using pfit_control to enable
the panel fitter. This has been changed from the below patch.

SQUASHME! - This patch should be squashed into the following
exising patch:
    Author: Arun R Murthy <arun.r.murthy@intel.com>
    Change-Id: I92651ed84d9d3f85fdcabbadd9cbb2f565483e11
    SQUASHME! [VPG]: drm/i915: Enable panel fitter only on HWC request

Change-Id: Iabd4e3684627d75cd6bfeeddc07958f9d9fac4c6
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-42114
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
## Don't forget to add a JIRA URL to the Tracked-On metadata!
## https://wiki.ith.intel.com/display/CACTUS/Patch+requirements+check
## url should be in the form:
## Tracked-On: https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
## Several Tracked-On lines are allowed
## you can fully prefill your Tracked-On commits using environment variable:
## export CURRENT_TRACKED_ON=https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
##
## Don't forget to add a JIRA URL to the Tracked-On metadata!
## https://wiki.ith.intel.com/display/CACTUS/Patch+requirements+check
## url should be in the form:
## Tracked-On: https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
## Several Tracked-On lines are allowed
## you can fully prefill your Tracked-On commits using environment variable:
## export CURRENT_TRACKED_ON=https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
##
---
 drivers/gpu/drm/i915/intel_sprite.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_sprite.c b/drivers/gpu/drm/i915/intel_sprite.c
index 70bf13c..f6bf42c 100644
--- a/drivers/gpu/drm/i915/intel_sprite.c
+++ b/drivers/gpu/drm/i915/intel_sprite.c
@@ -1535,7 +1535,7 @@ intel_update_plane(struct drm_plane *plane, struct drm_crtc *crtc,
 	min_scale = intel_plane->can_scale ? 1 : (1 << 16);
 
 	if (IS_VALLEYVIEW(dev) && intel_crtc->scaling_src_size &&
-		intel_crtc->config.gmch_pfit.control) {
+		(intel_crtc->pfit_control & PFIT_ENABLE)) {
 		clip.x2 = ((intel_crtc->scaling_src_size >>
 				SCALING_SRCSIZE_SHIFT) &
 				SCALING_SRCSIZE_MASK) + 1;
-- 
1.9.1

